File: Advanced Python Course (Python)/L2/zad2.py
# Ksawery Plis
# Lista 2 Zadanie 2

def pierwiastek(n):
    suma = 0
    for i in range(n):
        suma += (2*i-1)
        if (suma >= n):
            return (i-1)


print(pierwiastek(42))

File: Selected Elements of Software Design Practice (JavaScript, SQL)/L1/zad2/zad2.js
for (var i = 1; i <= 100000; i++) {
    var div = true;
    var x = i;
    var suma = 0;
    while (x > 0) {
        if (i % (x % 10) != 0) {
            div = false;
            break;
        }
        suma += (x % 10);
        x = (x - (x % 10)) / 10;
    }

    if (i % suma == 0 && div == true) console.log(i);

}